_posts/2018-08-15-JS-SDK.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title: The JS SDK hits version 2.0
+date: 2018-08-15 15:16 -0400
+comments: true
+author: flovilmart
+categories: [announcements, javascript, sdk]
+---
+
+The summer has been really busy and we've dusted out the JS SDK and striped large parts of it's heritage. `Parse.Promises` and `Backbone` style callbacks are definitely out.
+
+<!-- more -->
+
+## *Why the move now you may ask?*
+
+Well, we're also working hard on parse-server version 3.0, which revamps cloud code interface. The Cloud Code 'callbacks' have always been in the form: `response.success('yay');` and `response.error('oops');`.
+
+The `response` object is following the backbone style callback form, with `success` and `error` callbacks. The next generation of Cloud Code will get rid of this interface that belongs to another age. In 2018, `async` functions and `Promise` are part of the JavaScript language, are available in the LTS node runtime, and not only the leading edge.
+
+## *What should I do?*
+
+- If you're not using `Parse.Promise`'s with the JavaScript SDK, your code will break. You'll need to rewrite your code from the backbone style success/error callbacks to use promises.
+
+- If you're using `Parse.Promise`'s the migration should be minimal. `Parse.Promise` was compatible to a certain extent with the native `Promise` interface.
+
+In both cases, you'll need to spend some time making sure your code still work appropriately.
+
+## *It's a lot of work, should I really move to SDK 2.0?*
+
+Yes it is, but it will be a `red diff`, less lines, less code, best code.
+
+Take the following example:
+
+```js
+function loginAndGetFeed(username, password, callback) {
+  Parse.User.logIn(username, password, {
+    success: function(user) {
+      const query = new Parse.Query('Feed');
+      query.find({
+        success: function(feed) {
+          callback(null, feed);
+        },
+        error: function(error) {
+          callback(error);
+        }
+      })
+    },
+    error: function(error) {
+      callback(error);
+    }
+  })
+}
+```
+
+With the modern SDK you get the equivalent with just a fraction of the lines:
+
+```js
+async function loginAndGetFeed(username, password,) {
+  await Parse.User.logIn(username, password);
+  return await new Parse.Query('Feed').find();
```

Comment:
eslint doesn't like `return await ...` in this case you don't need the return.

Reply:
ah right you don't need `await`